core/src/main/java/aima/core/search/basic/support/BasicProblem.java

```diff
@@ -73,6 +75,16 @@ public BasicProblem(ActionsFunction<A, S> actionsFn, GoalTestPredicate<S> goalTe
 		this.stepCostFn = stepCostFn;
 	}
 
+	// BidirectionalProblem constructor(Reverse Problem)
```

Comment:
Move this to below the problem constructors (and ensure you have a blank line between method definitions). Again so that the order matches that of the book in terms of introduction to concepts.